Cannot compile with ghc-9.10.1

Compiling a downstream library using ghc-9.10.1, with `allow-newer: haskell-src-meta:template-haskell`, I run into:

```
[3 of 6] Compiling Language.Haskell.Meta.Syntax.Translate ( src/Language/Haskell/Meta/Syntax/Translate.hs, dist/build/Language/Haskell/Meta/Syntax/Translate.o, dist/build/Language/Haskell/Meta/Syntax/Translate.dyn_o )
src/Language/Haskell/Meta/Syntax/Translate.hs:801:17: error: [GHC-83865]
    • Couldn't match expected type ‘TH.Dec’
                  with actual type ‘TH.Name -> TH.Dec’
    • Probable cause: ‘TH.InfixD’ is applied to too few arguments
      In the expression: TH.InfixD (TH.Fixity fixity dir) (toName op)
      In the first argument of ‘map’, namely
        ‘(\ op -> TH.InfixD (TH.Fixity fixity dir) (toName op))’
      In the expression:
        map (\ op -> TH.InfixD (TH.Fixity fixity dir) (toName op)) ops
    |
801 |     map (\op -> TH.InfixD (TH.Fixity fixity dir) (toName op)) ops
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/Language/Haskell/Meta/Syntax/Translate.hs:801:51: error: [GHC-83865]
    • Couldn't match expected type ‘TH.NamespaceSpecifier’
                  with actual type ‘TH.Name’
    • In the second argument of ‘TH.InfixD’, namely ‘(toName op)’
      In the expression: TH.InfixD (TH.Fixity fixity dir) (toName op)
      In the first argument of ‘map’, namely
        ‘(\ op -> TH.InfixD (TH.Fixity fixity dir) (toName op))’
    |
801 |     map (\op -> TH.InfixD (TH.Fixity fixity dir) (toName op)) ops
    |                                                   ^^^^^^^^^
```

--- a/haskell-src-meta/ChangeLog
+++ b/haskell-src-meta/ChangeLog
@@ -1,3 +1,6 @@
+0.8.14
+- Support for GHC 9.10 (by Troels Henriksen)
+
 0.8.13
 - Support for GHC 9.8 (by Vladislav Zavialov)
 

--- a/haskell-src-meta/haskell-src-meta.cabal
+++ b/haskell-src-meta/haskell-src-meta.cabal
@@ -9,7 +9,7 @@ author:             Matt Morrow
 copyright:          (c) Matt Morrow
 maintainer:         danburton.email@gmail.com
 bug-reports:        https://github.com/haskell-party/haskell-src-meta/issues
-tested-with:        GHC == 8.2.2, GHC == 8.4.4, GHC == 8.6.5, GHC == 8.8.4, GHC == 8.10.7, GHC == 9.0.2, GHC == 9.2.2, GHC == 9.4.1, GHC == 9.6.1, GHC == 9.8.1
+tested-with:        GHC == 8.2.2, GHC == 8.4.4, GHC == 8.6.5, GHC == 8.8.4, GHC == 8.10.7, GHC == 9.0.2, GHC == 9.2.2, GHC == 9.4.1, GHC == 9.6.1, GHC == 9.8.1, GHC == 9.10.1
 synopsis:           Parse source to template-haskell abstract syntax.
 description:        The translation from haskell-src-exts abstract syntax
                     to template-haskell abstract syntax isn't 100% complete yet.
@@ -22,7 +22,7 @@ library
                    haskell-src-exts >= 1.21 && < 1.24,
                    pretty >= 1.0 && < 1.2,
                    syb >= 0.1 && < 0.8,
-                   template-haskell >= 2.12 && < 2.22,
+                   template-haskell >= 2.12 && < 2.23,
                    th-orphans >= 0.12 && < 0.14
 
   hs-source-dirs:  src

--- a/haskell-src-meta/src/Language/Haskell/Meta/Extensions.hs
+++ b/haskell-src-meta/src/Language/Haskell/Meta/Extensions.hs
@@ -367,6 +367,13 @@ fromExtension e = case e of
   TH.ExtendedLiterals                  -> Nothing
 #endif
 
+-- 2.22.0 ---------------------------------------
+
+#if MIN_VERSION_template_haskell(2,22,0)
+  TH.RequiredTypeArguments             -> Nothing
+  TH.ListTuplePuns                     -> Nothing
+#endif
+
   -- NB: when adding a case here, you may also need to update `toExtension`
 
 

--- a/haskell-src-meta/src/Language/Haskell/Meta/Syntax/Translate.hs
+++ b/haskell-src-meta/src/Language/Haskell/Meta/Syntax/Translate.hs
@@ -798,7 +798,11 @@ instance ToDecs (Exts.Decl l) where
   toDecs (Exts.InfixDecl l assoc Nothing ops) =
       toDecs (Exts.InfixDecl l assoc (Just 9) ops)
   toDecs (Exts.InfixDecl _ assoc (Just fixity) ops) =
+#if MIN_VERSION_template_haskell(2,22,0)
+    map (\op -> TH.InfixD (TH.Fixity fixity dir) TH.NoNamespaceSpecifier (toName op)) ops
+#else
     map (\op -> TH.InfixD (TH.Fixity fixity dir) (toName op)) ops
+#endif
    where
     dir = case assoc of
       Exts.AssocNone _  -> TH.InfixN

